app/controllers/foreman_ansible/api/v2/hostgroups_controller_extensions.rb

```diff
@@ -45,10 +47,63 @@ def multiple_play_roles
 
             render_message @result
           end
+
+          api :GET, '/hostgroups/:id/list_ansible_roles',
+              N_('Lists assigned Ansible roles')
+          param :id, :identifier, :required => true
+
+          def list_ansible_roles
+            find_resource
+
+            @result = {
+              :roles => @hostgroup.all_ansible_roles
+            }
+
+            render_message @result
+          end
+
+          api :POST, '/hostgroups/:id/ansible_roles',
+              N_('Assigns Ansible roles to a host group')
+          param :id, :identifier, :required => true
+          param :roles, Array, :required => true
+
+          def ansible_roles
+            find_resource
+
+            @hostgroup.ansible_roles = @roles - \
```

Comment:
My concern is that if the hostgroup parent gets removed , the children might end up without roles as an unintended action